webui/src/lib/onboarding/repoOnboardingService.tsx

```diff
@@ -0,0 +1,136 @@
+      </>
+    ),
+    cta: "Run import",
+    onClick: () =>
+      navigate(`/repositories/${currentRepo}/objects?importDialog=true`),
+    showStep: () => true, // objectStoreName !== "local",
+    isCompleted: async () => {
+      const repoBranches = await branches.list(currentRepo);
+      // when the API client is typed, we can remove the eslint-disable
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      return repoBranches.results.some((branch: any) =>
+        IMPORT_BRANCH_REGEX.test(branch.id)
+      );
+    },
+  },
+  {
+    id: "garbage-collection",
+    title: "Setup garbage collection",
+    description: (
+      <>
+        <p>Set up rules to automatically manage when data can be deleted from your repository</p>
+        <p>For more information about setting up and running garbage collection on your lakeFS repository, see the <a href="https://docs.lakefs.io/howto/garbage-collection-committed.html" target="_blank">documentation</a></p>
+      </>
+    ),
+    cta: "Configure",
+    onClick: () => navigate(`/repositories/${currentRepo}/settings/retention`),
+    showStep: () => true,
+    isCompleted: async () => {
+      try {
+        await retention.getGCPolicy(currentRepo);
+        return true;
+      } catch (e) {
+        return false;
+      }
+    },
+  },
+  {
+    id: "branch-protection",
+    title: "Set up branch protection",
+    description: (
+      <>
+        <p>You can protect important branches with branch protection rules.</p>
+        <p>Protected branches cannot be changed directly. Any changes are applied only via merging from other branches. This helps protect those branches from bugs in your code as well as human error.</p>
+
+      </>
+    ),
+    cta: "Configure",
+    onClick: () => navigate(`/repositories/${currentRepo}/settings/branches`),
+    showStep: () => true,
+    isCompleted: async () => {
+      const rules = await branchProtectionRules.getRules(currentRepo);
+      return rules.length > 0;
+    },
+  },
+  {
+    id: "hooks",
+    title: "Set up hooks",
+    description:
+      "Create simple hooks to validate new data meets certain requirements: schema validation, metadata registration, and more.",
+    cta: "Configure",
+    onClick: () =>
+      window.open("https://docs.lakefs.io/hooks/overview.html", "_blank"),
+    showStep: () => true,
+    isCompleted: async () => {
+      const objectList = await objects.list(currentRepo, defaultBranch, "");
+      return objectList.results.some(
+        // when the API client is typed, we can remove the eslint-disable
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        (object: any) => object?.path === HOOKS_PATH
+      );
```

Comment:
Same comment about listing having pages applies here, of course: this seems incorrect and also inefficient.  I would recommend listing objects with prefix HOOKS_PATH with limit 1 and looking at the result.

Reply:
Done